### PR TITLE
Serialize System components to HDF5

### PR DESCRIPTION
**Pull request by daniel-thom**
Objective: The simulations postprocessing code in PowerAnalytics.jl needs to have access to the systems used by the simulations. Those systems are not serialized if the user opts out of that feature. Ensure that these are always available.

This is the PowerSimulations portion of the solution. I added comparable branches in PowerSystems.jl and InfrastructureSystems.jl. Let me know if the interfaces here meet your needs and then I’ll open sibling PRs.

### CODE REVIEW

File: test/test_simulation_results.jl [hunk 1]
```diff
@@ -704,3 +730,26 @@ end
         test_simulation_results(file_path, export_path; in_memory = in_memory)
     end
 end
+
+@testset "Test simulation results with system from store" begin
```
Comment (daniel-thom) [inline test/test_simulation_results.jl:734]: The changes in this file are minor. I had to shuffle things around so that I could test the new code here.

**Review by GabrielKS — CHANGES_REQUESTED** [review]
Looks pretty good to me besides the comment about the `get_system!` kwarg.

**Review by jd-lara — APPROVED** [review]

**Comment by jd-lara** [comment]
@daniel-thom run the formatter to merge

**Comment by daniel-thom** [comment]
> @daniel-thom run the formatter to merge

Done

**Pull request merged**